modules/light-clients/08-wasm/types/vm.go

```diff
@@ -139,19 +144,72 @@ func wasmSudo[T ContractResult](ctx sdk.Context, clientStore storetypes.KVStore,
 	if err := json.Unmarshal(resp.Data, &result); err != nil {
 		return result, errorsmod.Wrap(ErrWasmInvalidResponseData, err.Error())
 	}
+
+	oldCodeHash := cs.CodeHash
```

Comment:
Since the code hash will actually not change...

```suggestion
	codeHash := cs.CodeHash
```